modules/mill-codegen-plugin/src/smithy4s/codegen/mill/Smithy4sModule.scala

```diff
@@ -108,15 +112,12 @@ trait Smithy4sModule extends ScalaModule {
 
   def smithy4sExternalCodegenIvyDeps: T[Agg[Dep]] = T {
     resolveDeps(transitiveIvyDeps)().flatMap { pathRef =>
-      val jarFile = new JarFile(pathRef.path.toIO)
-      val deps = Option(
-        jarFile
-          .getManifest()
-          .getMainAttributes()
-          .getValue(SMITHY4S_DEPENDENCIES)
-      ).toList.flatMap { listString =>
-        listString.split(",").toList.map(dep => ivy"$dep")
-      }
+      val deps = JarUtils
+        .extractJarManifestAttribute(pathRef.path.toIO, SMITHY4S_DEPENDENCIES)
+        .toList
+        .flatMap { listString =>
+          listString.split(",").toList.map(dep => ivy"$dep")
```

Comment:
`.split(",")` could be done inside JarUtils I think, that way we can reuse that

Reply:
because I named it `extractJarManifestAttribute` I decided to left it out (even if it was common)
that's because I can see it being used in a different context where the `key` won't be `SMITHY4S_DEPENDENCIES` and splitting would not make sense

Reply:
I just added an additional methods to reduce the duplication a little further.